sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -1479,4 +1481,20 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
         |FROM (SELECT '{"f1": "value1", "f2": 12}' json, 'hello' as str) test
       """.stripMargin), Row("value1", "12", 3.14, "hello"))
   }
+
+  // This test case is to verify a bug when making a new instance of LogicalRDD.
+  test ("SPARK-11633: HiveContext throws TreeNode Exception : Failed to Copy Node") {
```

Comment:
Thank you for sharing this! I will keep it in mind and follow this in the future. 

BTW, if you think it is necessary, I still can change it now. Thanks!